### ISSUE TITLE
Adding coordconv to unet leads to stange results

### ISSUE THREAD

**Issue by mrgloom** [titu1994/keras-coordconv#12]
I have synthetic experiment where on ground truth mask I have only 1 pixel (other are black) that represent center of ellipse:
https://github.com/mrgloom/random_code_samples/tree/master/DL/unet_loss_experiment

Using weighted binary cross entropy I get result similar to:
https://github.com/mrgloom/random_code_samples/blob/master/DL/unet_loss_experiment/unet_weighted_bce_loss.py
<img width="384" alt="screenshot 2018-12-09 at 22 18 36" src="https://user-images.githubusercontent.com/4003908/49701728-ef500b00-fc00-11e8-9973-536803317bb4.png">

But when I added coordconv layer on input at same number of epochs I get:
https://github.com/mrgloom/random_code_samples/blob/master/DL/unet_loss_experiment/unet_coord_conv.py
<img width="380" alt="screenshot 2018-12-09 at 22 17 28" src="https://user-images.githubusercontent.com/4003908/49701735-07c02580-fc01-11e8-97b3-7eede9174107.png">

i.e. first one is not perfect, but better then 2nd one.

Any other tricks can be applied to fix this issue?

**Comment by titu1994** [comment]
Hmm. What I can see is that the input image to the unet coordinate conv model doesn't have a gradient like the one above it. 

Even considering that to not be a significant factor, I don't have a clear idea of why adding coordinate a is normalized to 1 would so severely affect the output. 

Also, I might be wrong here, but for pixel segmentation tasks, isn't binary cross entropy loss summed rather than averaged accross the batch?
